target-gen/src/commands/test.rs

```diff
@@ -43,8 +43,13 @@ pub fn cmd_test(
 
     let file = File::open(Path::new(definition_export_path))?;
     probe_rs::config::add_target_from_yaml(file)?;
+    let file_handle = File::open(Path::new(definition_export_path))?;
+    let family: ChipFamily = serde_yaml::from_reader(file_handle)?;
+    // This assumes there is only one chip defined in the file, perhaps should be done for each family in the file.
+    // We need to get the chip name so that special startup procedure can be used. (matched on name)
```

Comment:
@Yatekii, also need some input here. It feels like a step forward, but the best case seems like running the test for all variants found. Is that feasible?

Reply:
`test` was intended specifically for developing flash algorithms, not for testing extracted ones.

How about we add an optional flag to select the variant via its name? :) We can also make an `--all` flag to test them all in a loop :)

Reply:
> test was intended specifically for developing flash algorithms, not for testing extracted ones.

I was using this to test my flash algorithm development. Problem was that the TI chips need the special startup sequence so I couldn't use `ALGORITHM_NAME` and then I ended up here :)

Reply:
I think that when developing a flash algorithm, its safe to assume that you only have one target per yaml, but it felt weird to hard code it, so now I iterate over them.

Reply:
Yeah, thats fair! :) I am not sure I understand the real problem here, but I am fine with the change :)

Reply:
So the problem I was trying to solve was that 
`probe_rs::Session::auto_attach(ALGORITHM_NAME, Permissions::new().allow_erase_all())?;` won't work for the TI chips or anything with a custom debug sequence since the [if statements in target.rs won't match](https://github.com/probe-rs/probe-rs/blob/master/probe-rs/src/config/target.rs#L117-L188).

For the TI device, this means the 2->4 pin JTAG swap wouldn't happen, and the test would fail.
What I did now is instead to read the target name from the `yaml` file instead of hard coding it. All of this is just to be sure that the `target` name is correct for the session creation here: `probe_rs::Session::auto_attach(target, Permissions::new().allow_erase_all())?;`

Then I thought, but what if they put multiple targets in the yaml, we should test them all. (e.g. multiple targets that should share the same flash algorithm), so that's why I added the iteration over families and devices in the input yaml.

Let me know if that makes sense :)

Reply:
Ah damn, makes sense! Thanks! I did nor consider that when I wrote the tool ...

Reply:
In general though it went super smooth to make the flash algorithm with these tests 😄